app/plugin_api.go

```diff
@@ -756,3 +756,38 @@ func (api *PluginAPI) UpdateBotActive(userId string, active bool) (*model.Bot, *
 func (api *PluginAPI) PermanentDeleteBot(userId string) *model.AppError {
 	return api.app.PermanentDeleteBot(userId)
 }
+
+func (api *PluginAPI) GetBotIconImage(userId string) ([]byte, *model.AppError) {
+	bot, err := api.app.GetBot(userId, true)
```

Comment:
A bot's user id is equivalent to the id of the user backing the bot -- so it should suffice to just pass `userId` through to `app.GetBotIconImage`, `app.SetBotIconImage`, etc.

However, I note that `app.GetBotIconImage` doesn't currently return an error if it's not a bot -- should it?

Reply:
In the api handler i was fetching the user and checking there as I need to use a user field, i thought fetching user twice might put extra load on the `get` api as it'll happen more frequent. But i think it should be ok. Better to be consistent across the 3 calls to verify if a user is a bot.